tests/test_intercept_parallel.js

```diff
@@ -61,4 +61,23 @@ describe('interception in parallel', () => {
     expect(results.sort()).to.deep.equal([200, 201, 418])
     expect(nock.isDone()).to.equal(true)
   })
+
+  it('provides the correct request instance on the Interceptor inside reply callbacks', async () => {
+    const seenFooHeaders = []
+    nock(origin)
+      .persist()
+      .get('/')
+      .reply(function () {
+        seenFooHeaders.push(this.req.headers.foo)
```

Comment:
Could this be clearer using a spy? You could assert called with `'A'`, and then a second assert called with `'B'`.

Reply:
Sorry, looks like I lost track of the PR.

To use Sinon for the assertions, I think it would look like the following. I'm not sure it makes the test any clearer.
```js
const fooHeadersStub = sinon.stub()

nock(origin)
  .persist()
  .get('/')
  .reply(function () {
    fooHeadersStub(this.req.headers.foo)
    return [200]
  })

await Promise.all([
  makeRequest({ headers: { foo: 'A' } }),
  makeRequest({ headers: { foo: 'B' } }),
])

expect(fooHeadersStub).to.have.callCount(2)
expect(fooHeadersStub).to.have.been.calledWithExactly('A')
expect(fooHeadersStub).to.have.been.calledWithExactly('B')
```

Reply:
@paulmelnikow I'd like to get this out. Any thoughts on preference between the current approach with an array vs Sinon?

Reply:
Just played around with this a bit. Personally I think the Sinon version is better. Though I do think this is just a style preference.

You could use `expect(fooHeadersStub).to.have.been.calledTwice()` in place of `callCount(2)`.